sdk/python/kfp/dsl/pipeline_config.py

```diff
@@ -73,5 +73,64 @@ def set_kubernetes_config(self,
 class PipelineConfig:
     """PipelineConfig contains pipeline-level config options."""
 
-    def __init__(self, workspace: Optional[WorkspaceConfig] = None):
+    def __init__(self,
+                 workspace: Optional[WorkspaceConfig] = None,
+                 semaphore_key: Optional[str] = None,
+                 mutex_name: Optional[str] = None):
         self.workspace = workspace
+        self._semaphore_key = semaphore_key
+        self._mutex_name = mutex_name
+
+    @property
+    def semaphore_key(self) -> Optional[str]:
+        """Get the semaphore key.
+
+        Returns:
+            Optional[str]: The semaphore key, or None if not set.
+        """
+        return self._semaphore_key
+
+    @semaphore_key.setter
+    def semaphore_key(self, value: str):
+        """Set the name of the semaphore to control pipeline concurrency.
+
+        The semaphore is configured via a ConfigMap. By default, the ConfigMap is
+        named "semaphore-config", but this name can be specified through the APIServer
+        deployment manifests using an environment variable named SEMAPHORE_CONFIGMAP_NAME.
+        If the environment variable is not specified, the default name "semaphore-config"
+        is used. The semaphore key is provided through the pipeline configuration.
+        If a pipeline has a semaphore, the backend maps the semaphore to the ConfigMap
+        using the key provided by the user.
+
+        Note: A pipeline can have both semaphores and mutexes (Argo Workflows v3.6+).
+        The workflow will block until all locks are available.
+
+        Args:
+            value (str): The key used to map to the ConfigMap.
+        """
+        if not value or not value.strip():
+            raise ValueError('Semaphore key is required and cannot be empty')
+        self._semaphore_key = value.strip()
+
+    @property
+    def mutex_name(self) -> Optional[str]:
+        """Get the mutex name.
+
+        Returns:
+            Optional[str]: The mutex name, or None if not set.
+        """
+        return self._mutex_name
+
+    @mutex_name.setter
+    def mutex_name(self, value: str):
+        """Set the name of the mutex to ensure mutual exclusion.
+
+        Note: A pipeline can have both semaphores and mutexes (Argo Workflows v3.6+).
+        The workflow will block until all locks are available.
+
+        Args:
+            value (str): Name of the mutex.
+        """
+        if not value or not value.strip():
+            raise ValueError('Mutex name is required and cannot be empty')
+        self._mutex_name = value.strip()
```

Comment:
Same comment as above
```suggestion
        self._mutex_name = (value and value.strip()) or None
```

Reply:
Done